### PR TITLE
Belos: add Tpetra pseudo_cg_indefinite test

### PR DESCRIPTION
**Pull request by tlamonthezie**
@trilinos/belos
@hkthorn

## Motivation
This PR creates a Tpetra version of the `pseudo_cg_indefinite` test.

### CODE REVIEW

**Review by hkthorn — APPROVED** [review]
LGTM

File: packages/belos/tpetra/test/BlockCG/test_pseudo_cg_indefinite.cpp [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+
+    for ( size_t k = 0; k < map->getLocalNumElements(); k++ )
+    {
+      GO gid = map->getGlobalElement(k);
+      const ST val = static_cast<ST> (2 * (gid-m) + 1);
+      A->insertGlobalValues(gid, tuple<GO> (gid), tuple<ST> (val));
+    }
+
+    if (proc_verbose)
+        std::cout << "TD: End Loop" << std::endl;
```
Comment (hkthorn) [inline packages/belos/tpetra/test/BlockCG/test_pseudo_cg_indefinite.cpp:128]: Remove these debugging statements.

File: packages/belos/tpetra/test/BlockCG/test_pseudo_cg_indefinite.cpp [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+    // construct diagonal matrix
+    const GO numGlobalElements = 100;
+    const int m = 4; // number of negative eigenvalues
+    // Create diagonal matrix with n-m positive and m negative eigenvalues.
+    RCP<const MAP> map = rcp(new MAP(numGlobalElements, 0, comm));
+
+    RCP<MAT> A = rcp( new MAT(map, 1) ); // DATA_Access::copy, , 1
+
+    if (proc_verbose)
+        std::cout << "TD: Start Loop" << std::endl;
```
Comment (hkthorn) [inline packages/belos/tpetra/test/BlockCG/test_pseudo_cg_indefinite.cpp:118]: Remove these debugging statements.

File: packages/belos/tpetra/test/BlockCG/test_pseudo_cg_indefinite.cpp [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+    }
+
+    if (proc_verbose)
+        std::cout << "TD: End Loop" << std::endl;
+
+    A->fillComplete();
+    TEUCHOS_ASSERT(A->isStorageOptimized());
+
+    if (proc_verbose)
+        std::cout << "TD: Storage is optimized" << std::endl;
```
Comment (hkthorn) [inline packages/belos/tpetra/test/BlockCG/test_pseudo_cg_indefinite.cpp:134]: Remove these debugging statements.

**Review by hkthorn — CHANGES_REQUESTED** [review]
Everything looks good except for some extraneous debugging statements that should be removed.

File: packages/belos/tpetra/test/BlockCG/test_pseudo_cg_indefinite.cpp (hunk 1 above)
Comment (tlamonthezie) [reply packages/belos/tpetra/test/BlockCG/test_pseudo_cg_indefinite.cpp]: Statements removed

File: packages/belos/tpetra/test/BlockCG/test_pseudo_cg_indefinite.cpp (hunk 3 above)
Comment (tlamonthezie) [reply packages/belos/tpetra/test/BlockCG/test_pseudo_cg_indefinite.cpp]: Statements removed

File: packages/belos/tpetra/test/BlockCG/test_pseudo_cg_indefinite.cpp (hunk 2 above)
Comment (tlamonthezie) [reply packages/belos/tpetra/test/BlockCG/test_pseudo_cg_indefinite.cpp]: Statements removed

**Comment by tlamonthezie** [comment]
> Everything looks good except for some extraneous debugging statements that should be removed.



> Everything looks good except for some extraneous debugging statements that should be removed.

Thanks @hkthorn for your review. Changes have been applied.

**Comment by cwschilly** [comment]
This PR needs retesting due to autotester failure last week

**Comment by cwschilly** [comment]
Hi @hkthorn @jwillenbring I think this PR's failure is still due to issues with the Autotester, rather than the code itself. Can we try testing again?

**Review by hkthorn — APPROVED** [review]
LGTM

**Pull request merged**